tests/phpunit/tests/ajax/CropImage.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/**
+ * Admin Ajax functions to be tested.
+ */
+require_once ABSPATH . 'wp-admin/includes/ajax-actions.php';
+
+require_once ABSPATH . 'wp-admin/includes/class-wp-filesystem-base.php';
+require_once ABSPATH . 'wp-admin/includes/class-wp-filesystem-direct.php';
+
+/**
+ * Class for testing ajax crop image functionality.
+ *
+ * @group ajax
+ */
+class Tests_Ajax_CropImage extends WP_Ajax_UnitTestCase {
+
+	/**
+	 * Tests that attachment properties are copied over to the cropped image.
+	 *
+	 * @ticket 37750
+	 *
+	 * @covers ::wp_ajax_crop_image
+	 */
+	public function test_it_copies_metadata_from_original_image() {
+
+		// Become an administrator.
+		$this->_setRole( 'administrator' );
+
+		$attachment = $this->create_attachment( true );
+		$this->prepare_post( $attachment );
+
+		// Make the request.
+		try {
+			$this->_handleAjax( 'crop-image' );
+		} catch ( WPAjaxDieContinueException $e ) {
+		}
+
+		$response = json_decode( $this->_last_response, true );
+		$this->validate_response( $response );
+
+		$cropped_attachment = get_post( $response['data']['id'] );
+		$this->assertInstanceOf( WP_Post::class, $cropped_attachment, 'get_post function must return an instance of WP_Post class' );
+		$this->assertNotEmpty( $attachment->post_title, 'post_title value must not be empty for testing purposes' );
+		$this->assertNotEmpty( $cropped_attachment->post_title, 'post_title value must not be empty for testing purposes' );
+		$this->assertSame( $attachment->post_title, $cropped_attachment->post_title, 'post_title value should be copied over to the cropped attachment' );
+		$this->assertSame( $attachment->post_content, $cropped_attachment->post_content, 'post_content value should be copied over to the cropped attachment' );
+		$this->assertSame( $attachment->post_excerpt, $cropped_attachment->post_excerpt, 'post_excerpt value should be copied over to the cropped attachment' );
+		$this->assertSame( $attachment->_wp_attachment_image_alt, $cropped_attachment->_wp_attachment_image_alt, '_wp_attachment_image_alt value should be copied over to the cropped attachment' );
+
+		wp_delete_attachment( $attachment->ID, true );
+		wp_delete_attachment( $cropped_attachment->ID, true );
```

Comment:
This logic will not be reached if a test fails. Instead move it to `tear_down()` which means the ID of each will need to become a property for `tear_down()` to have access.

Reply:
I've already tried to move this code to `tear_down`, and the attached images weren't deleted. I've just released (thanks to your comment) that I can fix it by calling `parent::tear_down()` at the very end of my `tear_down` method. 🤦

Otherwise, DB entities will be rolled back in the parent `tear_down` method and `wp_delete_attachment` method will not work.
Fixed in c55674c7b8e71c6043f2bd6aa35cf42d5c6d51be